xcdat/averager.py

```diff
@@ -397,62 +428,55 @@ def spatial_average(xvariable,
     Examples
     --------
     Import:
+
     >>> from xcdat.dataset import open_dataset
     >>> from xcdat.variable import open_variable
 
     Open dataset and dataarray:
+
     >>> filename = 'test.nc'
     >>> ds = open_dataset(filename)
     >>> tas = open_variable(ds, 'tas')
 
     Get global average time series:
+
     >>> ts_global = spatial_average(tas)
 
     Get time series in Nino 3.4 domain:
+
     >>> ts_n34 = spatial_average(tas,
                                  lat_bounds=[-5, 5],
                                  lon_bounds=[-170, -120])
 
     Get zonal mean time series:
+
     >>> ts_zonal = spatial_average(tas, axis=['lon'])
 
     Validate / fix bounds:
+
     >>> validated_regional_bounds = validate_region_lon_bounds(tas.bounds.lon,
                                                                regional_bounds)
-
-    Raises
-    ------
-    ValueError
-        Thrown if there is a conflict between specified bounds and the
-        dataarray bounds
     """
-    # Check to see if the region bounds are within the domain of the dataset
-    if lon_bounds:
-        # if not, guess the axis orientation (i.e., -180 to 180 or 0 to 360) of
-        # the dataarray axes and then convert the region bounds to that
-        # orientation
-        if not _validate_region_lon_bounds(xvariable.bounds.lon, lon_bounds):
-            if np.max(xvariable.lon) <= 180:
-                lon_bounds = _switch_lon_bounds(lon_bounds, totype='to180')
-            else:
-                lon_bounds = _switch_lon_bounds(lon_bounds, totype='to360')
-            # after region bounds are adjusted, check to ensure they are
-            # consistent with the dataarray axes
-            if not _validate_region_lon_bounds(xvariable.bounds.lon,
-                                               lon_bounds):
-                # REVIEW: Should we throw a more specific error?
-                raise ValueError('Attempted to fix specified longitude ' +
-                                 'bounds, but they are not within ' +
-                                 ' the longitude axis domain.')
+    # TODO: Is `lat_bounds` and `lon_bounds` only two floating points for the
+    # upper and lower bounds or None? Or can a <2 or >2 values be specified?
+    if type(axis) == str:
+        axis = [axis]  # type:ignore
+    for axes in axis:
+        if axes not in SUPPORTED_AXES:
+            raise ValueError(
+                "Incorrect `axis` argument. Supported axis include: "
+                f"{', '.join(SUPPORTED_AXES)}."
+            )
+
+    # TODO: Do we need to perform validation on region lat bounds too?
+    if lon_bounds is not None:
+        domain_lon_bounds = xvariable.bounds.get_bounds("lon")
+        lon_bounds = _validate_region_lon_bounds(lon_bounds, domain_lon_bounds)
```

Comment:
TODO: Do we need to perform validation on region lat bounds too?

Reply:
I don't think this is necessary. A longitude value of -90 and 270 are technically equivalent, but the spatial averaging calculation depends on the longitude convention used in the dataset. This bit of code is intended to ensure our code works with either convention. Latitude doesn't have this problem: it is always -90 to 90. We could validate that user-specified bounds are in this range, but I don't think that is necessary.